src/main/sarl/fr/utbm/info/ia51/crowdproject/motion/MotionAlgorithms.sarl

```diff
@@ -43,9 +47,23 @@ class KinematicWanderAlgorithm implements WanderAlgorithm {
  */
 class SteeringWanderAlgorithm implements WanderAlgorithm {
 
-	override run(emitter : UUID, currentPosition : Vector2) : MotionQuery {
-		val query = new MotionQuery(emitter, MotionBehavior::STEERING)
-		// TODO
-		return query
+	override run(emitter : UUID, currentPosition : Vector2, orientation : double, velocity : Vector2) : MotionQuery {
+		/* Calculate the circle center */
+		var center = velocity
+		center.magnitude = WANDER_CIRCLE_DISTANCE
+
+		/* Calculate the displacement and its angle */
+		var displacement = velocity
+		displacement.magnitude = WANDER_CIRCLE_RADIUS
+		var wanderingAngle = WANDER_MAX_ROTATION * binomialRandom
```

Comment:
We need it at each step in this algorithm so i think it's better to avoid a recalculation : we already have it